### PR TITLE
 feat(auth): implement OTP-based email verification flow

### PR DESCRIPTION
**Pull request by OlajideTechie**
**Description:**

This PR implements the end-to-end email verification flow for users registering via the LOCAL provider. It ensures account security by requiring a 6-digit OTP before issuing authentication tokens.

**Key Changes:**

**OTP Infrastructure:** Added EmailOtpEntity for persistence and OtpUtil for secure numeric code generation.

**Service Layer:** Implemented sendVerificationOtp and verifyEmail in LocalAuthService using TypeORM transactions to ensure data consistency.

**Email Integration:** Enhanced EmailService to support multiple providers (Gmail, SMTP) and added MailDev to the local Docker environment for sandbox testing.

**API Endpoints:**

Updated signupUser to trigger OTP delivery.
Added POST /api/v1/auth/verify-email to validate codes and finalize user activation.
Quality Assurance: Added comprehensive unit tests for both service logic and controller behavior, covering success and failure edge cases (expired/invalid OTPs).

**Testing:**

Run docker compose --profile dev up -d to start MailDev.
Register a user via /auth/signup.
Check http://localhost:1080 for the OTP.
Verify via /auth/verify-email.

image

**Response After Email Verification**

```
{
  "status": "success",
  "message": "Email verified",
  "data": {
    "user": {
      "id": "2dcfb82d-821b-4935-9279-3dd166d36098",
      "first_name": null,
      "last_name": null,
      "email": "janeKenned1013@yopmail.com",
      "avatar": null,
      "provider": "local",
      "role": null,
      "profile_completed": false,
      "state": null,
      "country": null,
      "linkedin_profile": null
    }
  },
  "tokens": {
    "access_token": ACCESS_TOKEN,
    "refresh_token": REFRESH_TOKEN
  }
}

```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -134,26 +134,64 @@ GOOGLE_CLIENT_ID=your-google-client-id
 GOOGLE_CLIENT_SECRET=your-google-client-secret
 LINKEDIN_CLIENT_ID=your-linkedin-client-id
 LINKEDIN_CLIENT_SECRET=your-linkedin-client-secret
-BASE_URL=https://localhost:8000
+BASE_URL=http://localhost:5000
```
Comment (AdeGneus) [inline README.md]: Any reason for changing this from port `8000` to `5000`?

Comment (AdeGneus) [reply README.md]: The `https` was used during testing of the LinkedIn OAuth but was later removed. Good job cleaning this part up but port `5000` contradicts `8000` that has been widely used in the env

Comment (OlajideTechie) [reply README.md]: Thanks for pointing it out. This must have been a typo, will update it

**Review by OlajideTechie — COMMENTED** [review]
Readme has been updated

**Review by AdeGneus — APPROVED** [review]
